### PR TITLE
feat: Weapons Station

### PR DESCRIPTION
**Pull request by joachimboyden**
Created a weapons station prefab (by default sells a shotgun) and a shotgun variant. 
Since there is no point system yet, I've made both the weapon and the ammo free for now. I've also made the ammo refill all your weapons ammo, but not the clip its self.

There are no Game Scene changes, so to test, just drag the prefab in. 
Also created a Interactable class, it currently only takes a radius of interaction and it's materials.

### CODE REVIEW

**Review by JDBryan — CHANGES_REQUESTED** [review]
If I buy a shotgun from the weapon station and then lose the shotgun by picking up another weapon, the weapon station will still only allow me to buy ammo rather than the expected behaviour of purchasing a weapon

**Comment by joachimboyden** [comment]
Fixed your problem with dropping weapons and some other bugs I noticed. I also dragged the shotgun station into the game scene and deleted the shotgun that was on the ground.

File: Assets/Scripts/Interactable.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    [HideInInspector] public bool selected;
+    public float selectionRadius;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        BaseStart();
+    }
+
+    public void BaseStart(){
```
Comment (JDBryan) [inline Assets/Scripts/Interactable.cs]: Rename function to Init

File: Assets/Scripts/WeaponStation.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public override void Interact(){
+        if (this.weaponOnSale == true){
+            player.PickupWeapon(this.weapon);
+            this.weaponOnSale = false;
+            this.restockInProgress = true;
+            animator.SetBool("Reload", true);
+        }
+        if (this.weaponOnSale == false & this.restockInProgress == false){
```
Comment (JDBryan) [inline Assets/Scripts/WeaponStation.cs]: Feels like you should maybe be using an else if here?

**Review by JDBryan — APPROVED** [review]
Approved with a couple of small comments

**Pull request merged**